### DISCUSSION TITLE
Texture to byte[]

### DISCUSSION THREAD

**Discussion by JackRumboll** [StereoKit/StereoKit#780] [General]
Can the texture be turned into a byte array to stream over a network? 
Do you know how I can accomplish this? I'm trying to stream screen capture data over the network as bytes.
I'm using the [DesktopDuplicator](https://github.com/maluoi/SKDesktopDuplicator) project to get a visual of the screen and now I just want to turn it into bytes to stream it.

**Comment by maluoi** [comment]
In this case, `DesktopDuplicator` relies on `Vortice.Direct3D11`, so you can always use D3D for fine control over extracting color data optimally. StereoKit does have [`Tex.GetColors`](https://stereokit.net/Pages/StereoKit/Tex/GetColors.html), but I'm not certain if it works well in this context or not, there's a lot of rules about pulling texture data into CPU side memory.

Streaming a whole desktop screen is a lot of work for computers and needs to be done carefully, so if you can get it compressed before moving it off the GPU, that'd be the ideal situation!
